#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.0016223110292997148 -0.9255016812340858 -0.378740024475635 3.149743908786134
  position 0.7669472845739005 18.494250476131626 -19.418282752971866
}
TexturedBackground {
}
TexturedBackgroundLight {
}
UnevenTerrain {
  translation 0.82 0.62 -2.06
  rotation 0 1 0 4.692820414042842e-06
  name "virtual garden"
  size 10 0 10
  appearance SandyGround {
    colorOverride 0 0.517906 0.133974
    textureTransform TextureTransform {
      scale 50 50
    }
    IBLStrength 3
  }
}
Apple {
  hidden linearVelocity_0 0.05891478393914201 1.0332563312157721e-15 -1.8047123300062195e-12
  hidden angularVelocity_0 -3.609424620567673e-11 2.1363406837746684e-11 -1.1782956787828391
  translation 0.7308585511776984 0.6697645599999998 -2.183978862260788e-11
  rotation 2.1798542525494372e-11 3.541708619120437e-11 1 1.4094835878409864
}
Apple {
  hidden linearVelocity_0 3.1928447137784066e-13 1.696659629508244e-15 2.0666450199720334e-44
  hidden angularVelocity_0 4.133290039944063e-43 -3.0047948382826967e-42 -6.385689427581813e-12
  translation 0.7675610000032914 0.6697645599999997 -1.78929
  rotation 6.1248568571585574e-15 4.474980960000451e-13 -1 0.08765091923114961
  name "apple(1)"
}
Apple {
  hidden linearVelocity_0 1.7421161132455897e-40 1.696659629508244e-15 -1.7614079026141897e-12
  hidden angularVelocity_0 -3.522815805233377e-11 -1.7172505021320038e-34 -3.484242816263984e-39
  translation -1.29257 0.6697645599999997 -4.635680000018838
  rotation -4.296174710457103e-09 -1.879552367930291e-10 -1 0.0876509191653243
  name "apple(3)"
}
Apple {
  hidden linearVelocity_0 1.8771316972224805e-14 1.696659629508244e-15 -1.7614079026141897e-12
  hidden angularVelocity_0 -3.522815805233377e-11 1.6267583504470196e-30 -3.754277397216802e-13
  translation 0.6161829999995831 0.6697645599999997 -5.673820000018838
  rotation -4.296174710866864e-09 -1.879552368102612e-10 -1 0.08765091915695956
  name "apple(4)"
}
Apple {
  hidden linearVelocity_0 0 1.696659629508244e-15 0
  translation 0.712857 0.6697645599999997 -3.95479
  rotation 6.124856847026981e-15 4.4749809633609983e-13 -1 0.0876509191653243
  name "apple(5)"
}
Apple {
  hidden linearVelocity_0 0 1.696659629508244e-15 0
  translation 2.63312 0.6697645599999997 -4.64346
  rotation 6.124856847026981e-15 4.4749809633609983e-13 -1 0.0876509191653243
  name "apple(6)"
}
Apple {
  hidden linearVelocity_0 4.2359964107524597e-41 1.696659629508244e-15 -8.807039513070948e-13
  hidden angularVelocity_0 -1.7614079026166886e-11 -4.2930485366975035e-35 -8.471992821315467e-40
  translation 2.23899 0.6697645599999997 -3.053150000009419
  rotation -2.148084292800128e-09 -9.375386934834649e-11 -1 0.0876509191653243
  name "apple(7)"
}
Apple {
  hidden linearVelocity_0 3.582982913953443e-15 1.696659629508244e-15 -8.807039513070948e-13
  hidden angularVelocity_0 -1.7614079026166886e-11 3.1118039964973566e-30 -7.166059181490352e-14
  translation -0.277441999999951 0.6697645599999997 -3.408660000009419
  rotation -2.148084292776652e-09 -9.375386934766462e-11 -1 0.08765091916628166
  name "apple(2)"
}
PicketFence {
  translation 5.22 0.48 -6.6
  rotation 0 1 0 3.14159
  name "picket fence(2)"
  numberOfSegments 4
}
PicketFence {
  translation -3.31 0.35 1.86
  name "picket fence(12)"
  numberOfSegments 5
}
PicketFence {
  translation 5.18 0.52 1.86
  rotation 9.58979e-09 -1 9.58976e-09 -1.5707953071795862
  name "picket fence(5)"
  numberOfSegments 5
}
PicketFence {
  translation -3.26 0.46 -6.65
  name "picket fence(8)"
}
PicketFence {
  translation -3.25 0.41 1.8
  rotation 0 1 0 1.5708
  name "picket fence(9)"
  numberOfSegments 5
}
PalmTree {
  translation -2.01 0 0.14
}
Oak {
  translation 3.68 0 -3.32
}
Oak {
  translation 3.43 0 0.63
  name "oak tree(1)"
}
Oak {
  translation 3.71 0 -5.28
  name "oak tree(2)"
}
Oak {
  translation -1.97 0 -5.64
  name "oak tree(3)"
}
Wheelbarrow {
  translation 4.34846 0.56 -1.97468
  rotation 0 1 0 1.5708
}
DogHouse {
  translation 3.61 0.65 -0.79
  rotation 0 1 0 1.5708
}
TableWithUmbrella {
  translation 0.55 0.48 0
}
Robot {
  translation -0.450817 0.7 -2.52828
  rotation 0 -1 0 6.53317e-06
  scale 1.5 1.5 1.5
  children [
    DEF base_link Shape {
      appearance PBRAppearance {
        baseColor 0.0659342 0.501793 0.00683604
        metalness 0
      }
      geometry Box {
        size 0.28 0.055 0.15
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor 0.05 0 0.09
      }
      device [
        RotationalMotor {
          name "front_right_wheel"
        }
      ]
      endPoint Solid {
        translation 0.04999634683682081 -2.094717650386603e-06 0.0899897142498077
        rotation 0.9779513156757368 0.1476091067171582 0.14772533899862275 1.592289000087949
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 0.8 0.5 0
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.055
            }
          }
        ]
        name "front_right_wheel"
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor -0.08 0 0.09
      }
      device [
        RotationalMotor {
          name "rear_right_wheel"
        }
      ]
      endPoint Solid {
        translation -0.08000587983417132 4.035589766518507e-06 0.08999957846809874
        rotation 0.9779513220677499 0.14760844352270006 0.1477259593529378 1.5922867322353302
        children [
          USE wheel_shape
        ]
        name "rear_right_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor 0.05 0 -0.09
      }
      device [
        RotationalMotor {
          name "front_left_wheel"
        }
      ]
      endPoint Solid {
        translation 0.04998022925514414 -8.461193206896541e-06 -0.08999478718007552
        rotation 0.9724219821858224 0.16485048945089376 0.16498425588399962 1.5979567954753642
        children [
          USE wheel_shape
        ]
        name "front_left_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor -0.08 0 -0.09
      }
      device [
        RotationalMotor {
          name "rear_left_wheel"
        }
      ]
      endPoint Solid {
        translation -0.08001673063184697 -1.2879564129989755e-05 -0.08999321361175705
        rotation 0.9724221603353507 0.16485004746572918 0.16498364749052064 1.5979562588572407
        children [
          USE wheel_shape
        ]
        name "rear_left_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
      }
    }
    DistanceSensor {
      translation 0.14 -0.0275 0.075
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ir_ext_right"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
    }
    DistanceSensor {
      translation 0.14 -0.0275 0.025
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ir_mid_right"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
    }
    DistanceSensor {
      translation 0.14 -0.0275 -0.075
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ir_ext_left"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
    }
    DistanceSensor {
      translation 0.14 -0.0275 -0.025
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ir_mid_left"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
    }
    DistanceSensor {
      translation 0.12 -0.0275 0.075
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ir_right"
      boundingObject USE ir_shape
      physics USE ir_physics
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
    }
    DistanceSensor {
      translation 0.12 -0.0275 -0.075
      rotation 0 1 0 1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ir_left"
      boundingObject USE ir_shape
      physics USE ir_physics
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
    }
    Camera {
      translation 0.14 0 0
      rotation 0 1 0 -1.57
      children [
        DEF camera_shape Shape {
          appearance PBRAppearance {
            baseColor 1 0.999969 0.999985
            metalness 0
          }
          geometry Cylinder {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "front_camera"
      boundingObject USE camera_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation -0.025 0.05 -0.065
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.2 0.1 0.01
          }
        }
      ]
      name "basket"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation -0.025 0.05 0.065
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.2 0.1 0.01
          }
        }
      ]
      name "basket(1)"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation -0.13 0.05 0
      rotation 0 1 0 1.57
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.14 0.1 0.01
          }
        }
      ]
      name "basket(2)"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation 0.08 0.05 0
      rotation 0 1 0 1.57
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.14 0.1 0.01
          }
        }
      ]
      name "basket(3)"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
  ]
  name "GardenRobo"
  boundingObject USE base_link
  physics Physics {
  }
  controller "my_controller"
}
SwingCouch {
  translation -1.88 0 -2.76
  rotation 0 1 0 3.14159
}
PottedTree {
  translation -0.6 0.62 -1.87
}
PottedTree {
  translation -0.6 0.62 -5.23
  rotation 0 1 0 -1.5707953071795862
  name "potted tree(2)"
}
PottedTree {
  translation 0.62 0.62 -3.32
  rotation 0 1 0 -1.5707953071795862
  name "potted tree(4)"
}
PottedTree {
  translation 1.81 0.61 -5.16
  rotation 0 1 0 -1.5707953071795862
  name "potted tree(3)"
}
PottedTree {
  translation 1.91071 0.61 -1.73929
  rotation 0 -1 0 5.30716e-06
  name "potted tree(1)"
}
